RAGchain/benchmark/dataset/eli5.py

```diff
@@ -0,0 +1,109 @@
+        self.qa_data = load_dataset(self.file_path + "-qa")['train'].to_pandas()
+        self.ingest_data = load_dataset(self.file_path + "-document")['train'].to_pandas()
+
+        default_metrics = self.retrieval_gt_metrics + self.answer_gt_metrics + self.answer_passage_metrics
+        support_metrics = default_metrics + self.retrieval_gt_ragas_metrics + self.retrieval_no_gt_ragas_metrics \
+                          + self.answer_no_gt_ragas_metrics
+
+        if metrics is not None:
+            # Check if your metrics are available in evaluation datasets.
+            for metric in metrics:
+                if metric not in support_metrics:
+                    raise ValueError("You input metrics that this dataset evaluator not support.")
+            using_metrics = list(set(metrics))
+        else:
+            using_metrics = default_metrics
+
+        super().__init__(run_all=False, metrics=using_metrics)
+
+        self.eval_size = evaluate_size
+        self.run_pipeline = run_pipeline
+
+        if evaluate_size is not None and len(self.qa_data) > evaluate_size:
+            self.qa_data = self.qa_data[:evaluate_size]
+
+    def ingest(self, retrievals: List[BaseRetrieval], db: BaseDB, ingest_size: Optional[int] = None):
+        """
+        Ingest dataset to retrievals and db.
+        :param retrievals: The retrievals that you want to ingest.
+        :param db: The db that you want to ingest.
+        :param ingest_size: The number of data to ingest. If None, ingest all data.
+        """
+        ingest_data = deepcopy(self.ingest_data)
+        if ingest_size is not None:
+            # ingest size must be larger than evaluate size.
+            if ingest_size >= self.eval_size:
+                ingest_data = ingest_data[:ingest_size]
+            else:
+                raise ValueError("ingest size must be same or larger than evaluate size")
+
+        # Create passages.
+        passages = ingest_data.apply(self.__make_passages, axis=1).tolist()
+
+        for retrieval in retrievals:
+            retrieval.ingest(passages)
+        db.create_or_load()
+        db.save(passages)
+
+    def evaluate(self, **kwargs) -> EvaluateResult:
+
+        return self._calculate_metrics(
+            questions=self.qa_data['question'].tolist(),
+            pipeline=self.run_pipeline,
+            retrieval_gt=self.qa_data.apply(lambda row: [uuid.UUID(row['doc_id'])], axis=1).tolist(),
+            answer_gt=self.qa_data.apply(lambda row: [row['goldenAnswer']], axis=1).tolist(),
+            **kwargs
+        )
+
+    def __make_passages(self, row):
+
+        return Passage(
+            id=row['doc_id'],
+            content=row['document'],
+            filepath=self.file_path,
+            metadata_etc={
+                'dataset_creator': 'Pakulski',
+                'dataset_url': 'https://huggingface.co/datasets/Pakulski/ELI5-test',
+                'dataset_name': 'ELI5-test',
+                'dataset_size': '1.3GB',
```

Comment:
Why you have to add this metadata? It seems useless for benchmarking. And they know where this dataset is come from. Right?
Just put annotation about dataset source.